packages/http-server/src/serializers.ts

```diff
@@ -34,3 +45,25 @@ export default [
     },
   },
 ];
+
+export const optionallySerializeAndSend = (
+  reply: FastifyReply<ServerResponse>,
+  output: { headers?: Dictionary<string, string>; body?: string | object },
+  respSerializers: Serializer[],
+) => {
+  const contentType = output.headers && output.headers['Content-type'];
+  const serializer = respSerializers.find((s: Serializer) => s.test(contentType || ''));
+
+  const data = serializer && output.body ? serializer.serialize(output.body) : output.body;
+  const isInUTF8 = contentType && contentType.includes('charset=utf-8');
+
+  if (serializer) {
+    reply.serializer(serializer.serialize);
+  }
+
+  if (contentType && contentType.includes('application/json')) {
+    reply.header('content-type', isInUTF8 ? contentType : contentType + '; charset=utf-8');
```

Comment:
This code is forcing `'; charset=utf-8')` onto _all_ responses even when nothing anywhere in the OpenAPI document or anywhere in the HTTP request or anywhere has suggested that we should. Could you explain your reasoning for adding this? Unless there is a compelling reason to add this (which I cannot think of¡) then we should not be adding this.

Reply:
So, it was Fastify that was adding the charset. All of the responses to requests that asked for `application/json` were serialized with the default Fastify serializer. This serializer was adding this encoding. In the cases of the adjusted tests files, `; charset=utf-8` was missing. Why? Because of `-H "accept: application/xml, application/json"`. So Fastify sees this and decides to serialize the response with https://github.com/stoplightio/prism/blob/a71e853f3dcba1d464f4cb1a39d6c7daf27118dc/packages/http-server/src/serializers.ts#L29, hence no `; charset=utf-8` (because it no longer used the default serializer). And why did this even work? Because of https://github.com/stoplightio/prism/blob/a71e853f3dcba1d464f4cb1a39d6c7daf27118dc/packages/http-server/src/serializers.ts#L31. So.. we had a string from the test file, so the XML serializer was not really serializing it.